HHS/Head-Start-TTADP
# [PROD] TTAHUB-3329 Fix view of TR event after restructure

## Description of change

**TTAHUB-3329:** Fixes for updates to view TR event after restructure.
**TTAHUB-2589** xml2json has been removed in favor of xml2js. topo and joi have been replaced by @hapi/topo and @hapi/joi.
**TTAHUB-3046** Adding a hook to populate the onAR and onApprovedAR flags when the merged goal is placed on the ActivityReportGoal records.


## How to test

**TTAHUB-3329:** View an event make sure the new display reflects the restructure.
**TTAHUB-2589** Verify tests pass.
**TTAHUB-3046** Merge goals that have been on an AR, the resulting goal has flags set correctly

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3329
* https://jira.acf.gov/browse/TTAHUB-2589
* https://jira.acf.gov/browse/TTAHUB-3046


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
